app/ts/McdaApp/Workspace/CurrentTab/Preferences/PreferencesWeights/PreferencesWeightsTable/preferencesWeightsTableUtil.ts

```diff
@@ -0,0 +1,105 @@
+import IChangeableValue from 'app/ts/interface/IChangeableValue';
+import _ from 'lodash';
+
+export function buildImportances(
+  weights: Record<string, number>
+): Record<string, number> {
+  const highestWeight = _.max(_.values(weights));
+  return _.mapValues(weights, (weight: number) =>
+    Math.round((weight / highestWeight) * 100)
+  );
+}
+
+export function calculateRankings(
+  weights: Record<string, number>
+): Record<string, number> {
+  const weightRankCriterionIds = getCriterionWeightAndRank(weights);
+  return getCorrectRanks(weightRankCriterionIds);
+}
+
+interface IWeightRankCriterionId {
+  rank: number;
+  weight: number;
+  criterionId: string;
+}
+
+function getCriterionWeightAndRank(
+  weights: Record<string, number>
+): Record<string, IWeightRankCriterionId> {
+  return _(weights)
+    .map((weight, criterionId) => {
+      return {weight, criterionId};
+    })
+    .orderBy('weight', 'desc')
+    .map((value, index) => {
+      return {...value, rank: index + 1};
+    })
+    .keyBy('criterionId')
+    .value();
+}
+
+function getCorrectRanks(
```

Comment:
maybe something like 'mergeSameWeightRanks' or something? 'Correct' is a bit vague